### DISCUSSION TITLE
Voice or Text-Based Trading

### DISCUSSION THREAD

**Discussion by lucaborella89** [finos/traderX#186] [Ideas]
Allow traders to place orders, query market data, and manage their portfolios using natural language commands (e.g., "Buy 100 shares of AAPL" or "What's the current price of gold?").

Main Flow:

1. Command Input: The trader provides a natural language command through voice or text input.
Examples:
"Buy 100 shares of AAPL at market price."
"Sell 50 shares of GOOG when the price reaches $3000."
"What's the current price of BTC?"
"Show me my portfolio performance."
2. Command Interpretation: The LLM interprets the command, extracting relevant information like the action (buy/sell), asset symbol, quantity, price, and any additional conditions.

**Comment by lucaborella89** [comment]
@tenatzen - any interest from Zenith to join the event and work on this use case?

**Reply by tenatzen** [reply]
Yes, we will table this on the upcoming brain trust (July 11th).

I recommend we also loop in the team behind [Voxura](https://github.com/finos-labs/voxura) that may already have a working solution for this.
